File: leetcode/php/0022-generate-parentheses.php
<?php

/**
 * 22. Generate Parentheses
 * https://leetcode.com/problems/generate-parentheses/
 */

class Solution
{
    /**
     * @param Integer $n
     * @param String $s
     * @return String[]
     */
    function generateParenthesis($n, $s = '()')
    {
        if ($n == 1) return [$s];
        for ($i = 1; $i <= strlen($s); $i++)
            $result = array_merge($result ?? [], $this->generateParenthesis($n - 1, substr($s, 0, $i) . '()' . substr($s, $i)));
        return array_unique($result);
    }
}

File: leetcode/swift/0022-generate-parentheses.swift
/**
 * 22. Generate Parentheses
 * https://leetcode.com/problems/generate-parentheses/
 */

class Solution {
    func generateParenthesis(_ n: Int, _ s: String = "()") -> [String] {
        if n == 1 {
            return [s]
        }
        var result: [String] = []
        for i in 1...s.length {
            let index = s.index(s.startIndex, offsetBy: i)
            result += self.generateParenthesis(n - 1, s[..<index] + "()" + s[index...])
        }
        return Array(Set(result))
    }
}